geode-core/src/main/java/org/apache/geode/cache/PartitionedRegionVersionException.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.cache;
+
+/**
+ * Indicates a failure to perform an operation on a Partitioned Region due to
+ * server versions not meeting requirements.
+ *
+ * @since GemFire 5.1
+ */
+public class PartitionedRegionVersionException extends CacheRuntimeException {
+  private static final long serialVersionUID = -3004093739855972548L;
+
+  public PartitionedRegionVersionException() {
+    super("A server's version was too old to a partition region clear");
+  }
```

Comment:
The exception name seems a bit miss-leading to me, the "actual problem" is the version of the members on which the partition region is hosted, not the region itself. Maybe `UnsupportedVersionException`, `IncompatibleVersionException` or something similar?.
It would also be good to include at least the id or name of one of the members that's running an unsupported version within the exception message, it will make the user's troubleshooting experience a lot better.

Reply:
I see your point. I think it is very valid and worth changing. How about ServerVersionException or MemberVersionException?

Reply:
Or perhaps VersionMismatchException

Reply:
would be nice to indicate which server is the older version

Reply:
Will do.

Reply:
The exception message is worded awkwardly. Should this be

- "too old to do a partition region clear"?
- "too old to clear a partition region"?
- something else?

Reply:
I have changed the exception to ServerVersionMismatchException(members, feature name) it now produces a message "A server's [oldMember] version was too old for : Partitioned Region Clear"